Neos.ContentRepository.Core/Classes/Projection/ContentGraph/AbsoluteNodePath.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\Projection\ContentGraph;
+
+use Neos\ContentRepository\Core\NodeType\NodeTypeName;
+use Neos\ContentRepository\Core\SharedModel\Exception\AbsoluteNodePathIsInvalid;
+use Neos\ContentRepository\Core\SharedModel\Node\NodeName;
+
+/**
+ * An absolute node path, composed of the root node's type and a relative path from there.
+ *
+ * It describes the hierarchy path of a node to and including its root node in a subgraph.
+ * @api
+ */
+final class AbsoluteNodePath implements \JsonSerializable
+{
+    private function __construct(
+        public readonly NodeTypeName $rootNodeTypeName,
+        public readonly NodePath $path
+    ) {
+    }
+
+    public static function fromComponents(
+        NodeTypeName $rootNodeTypeName,
+        NodePath $path
+    ): self {
+        return new self(
+            $rootNodeTypeName,
+            $path
+        );
+    }
+
+    public static function patternIsMatchedByString(string $value): bool
+    {
+        if (!\str_starts_with($value, '/<')) {
+            return false;
+        }
+        if (!(\mb_strpos($value, '>') > 0)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static function fromString(string $value): self
+    {
+        if (!self::patternIsMatchedByString($value)) {
+            throw AbsoluteNodePathIsInvalid::becauseItDoesNotMatchTheRequiredPattern($value);
+        }
+        $pivot = \mb_strpos($value, '>') ?: 0; // pivot is actually > 0 due to the pattern check above
+        $nodeTypeName = NodeTypeName::fromString(\mb_substr($value, 2, $pivot - 2));
+        $path = \mb_substr($value, $pivot + 2) ?: '/';
+
+        return new self($nodeTypeName, NodePath::fromString($path));
+    }
+
+    public static function tryFromString(string $string): ?self
+    {
+        return self::patternIsMatchedByString($string)
+            ? self::fromString($string)
+            : null;
+    }
+
+    public function appendPathSegment(NodeName $nodeName): self
+    {
+        return new self(
+            $this->rootNodeTypeName,
+            $this->path->appendPathSegment($nodeName)
+        );
+    }
+
+    /**
+     * While all absolute node paths _have_ a root node type name,
+     * they _are_ root only if they point to exactly that root node and not one of its descendants
+     */
+    public function isRoot(): bool
+    {
+        return $this->path->value === '/';
```

Comment:
Imho this should be encapsuled by `NodePath`, so that this can be expressed as:

```suggestion
        return $this->path->isEmpty();
```